File: src/main/java/me/Oxygen/modules/render/Animation.java
package me.Oxygen.modules.render;

import me.Oxygen.event.Event;
import me.Oxygen.event.EventTarget;
import me.Oxygen.event.events.EventUpdate;
import me.Oxygen.modules.Category;
import me.Oxygen.modules.Module;
import me.Oxygen.modules.ModuleRegister;
import me.Oxygen.value.Value;

@ModuleRegister(name = "Animation", category = Category.RENDER)
public class Animation extends Module {
	
	public static Value<String> mode = new Value<String>("Animation", "Mode", 0);
	
	public Animation() {
		mode.addValue("Leakey");
		mode.addValue("Jello");
		mode.addValue("Sigma");
		mode.addValue("MeMe");
		mode.addValue("1.7");
	}
	
	@EventTarget(events = EventUpdate.class)
	public void onEvent(Event e) {
		if(e instanceof EventUpdate) {
		this.setDisplayName(mode.getModeAt(mode.getCurrentMode()));
		}
	}
	

}

File: src/main/java/cn/rainbow/oxygen/module/modules/render/Animation.kt
package cn.rainbow.oxygen.module.modules.render

import cn.rainbow.oxygen.module.Category
import cn.rainbow.oxygen.module.Module
import cn.rainbow.oxygen.module.ModuleInfo
import cn.rainbow.oxygen.module.setting.ModeValue

@ModuleInfo(name = "Animation", category = Category.Render, noSetEnable = true)
class Animation: Module() {
    val mode = ModeValue("Mode", "Normal", arrayOf("Normal", "Exhi", "Sigma", "1.7"))
}